logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/SendConfirmationUseCase.kt

```diff
@@ -0,0 +1,83 @@
+package com.wire.kalium.logic.feature.message
+
+import com.benasher44.uuid.uuid4
+import com.wire.kalium.logger.KaliumLogger
+import com.wire.kalium.logic.CoreFailure
+import com.wire.kalium.logic.data.conversation.ConversationRepository
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.message.Message
+import com.wire.kalium.logic.data.message.MessageContent
+import com.wire.kalium.logic.data.message.MessageRepository
+import com.wire.kalium.logic.data.message.receipt.ReceiptType
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.CurrentClientIdProvider
+import com.wire.kalium.logic.functional.Either
+import com.wire.kalium.logic.functional.flatMap
+import com.wire.kalium.logic.functional.fold
+import com.wire.kalium.logic.functional.onFailure
+import com.wire.kalium.logic.functional.onSuccess
+import com.wire.kalium.logic.kaliumLogger
+import com.wire.kalium.logic.sync.SyncManager
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.datetime.Clock
+
+/**
+ * This use case allows to send a confirmation type [ReceiptType.READ] or [ReceiptType.DELIVERED] accordingly to
+ * configurations criteria.
+ *
+ * - For 1:1 we take into consideration [ObserveReadReceiptsEnabled]
+ * - For group conversations we have to look for each group conversation configuration.
+ */
+internal class SendConfirmationUseCase internal constructor(
```

Comment:
Indeed! We came up with that exact name in the beginning but changed as we thought we needed to send both `READ` and `DELIVERED` , but we per current implementation whenever we receive a message on AR it already gets set as `DELIVERED` to the other party so it won't be needed here and it can indeed be renamed to just `SendReadConfirmationUseCase` in the near future in another PR.